frame/support/src/storage/mod.rs

```diff
@@ -811,16 +815,292 @@ mod private {
 
 	impl<T: Encode> Sealed for Vec<T> {}
 	impl<Hash: Encode> Sealed for Digest<Hash> {}
+	impl<T: Value, S: Get<u32>> Sealed for BoundedVec<T, S> {}
 }
 
 impl<T: Encode> StorageAppend<T> for Vec<T> {}
 impl<T: Encode> StorageDecodeLength for Vec<T> {}
 
-/// We abuse the fact that SCALE does not put any marker into the encoding, i.e.
-/// we only encode the internal vec and we can append to this vec. We have a test that ensures
-/// that if the `Digest` format ever changes, we need to remove this here.
+/// We abuse the fact that SCALE does not put any marker into the encoding, i.e. we only encode the
+/// internal vec and we can append to this vec. We have a test that ensures that if the `Digest`
+/// format ever changes, we need to remove this here.
 impl<Hash: Encode> StorageAppend<DigestItem<Hash>> for Digest<Hash> {}
 
+/// Marker trait for types `T` that can be stored in storage as `Vec<T>`.
+pub trait Value: FullCodec + Clone + sp_std::fmt::Debug + Eq + PartialEq {}
+impl<T: FullCodec + Clone + sp_std::fmt::Debug + Eq + PartialEq> Value for T {}
+
+/// A bounded vector.
+///
+/// It implementations for efficient append and length decoding, as with a normal `Vec<_>`, once put
+/// into a [`StorageValue`].
```

Comment:
I think it is more precise to say that the encoding is the same as the encoding of `Vec<T>`, because it can also be used in `StorageMap ` or anywhere.

Reply:
addressed.